### PR TITLE
[feat] explicitly set compilerOptions.hydratable to config.kit.ssr

### PR DESCRIPTION
**Pull request by dominikg**
### Before submitting the PR, please make sure you do the following
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

Until now vite-plugin-svelte defaults to `hydratable: true`, but that is going to change. 
See https://github.com/sveltejs/vite-plugin-svelte/pull/122

To avoid breaking, kit has to set hydratable explicitly.
vite-plugin-svelte also offers a new option `experimental.dynamicCompileOptions` that could be used to implement a more finegrained solution of which components need hydration. see https://github.com/sveltejs/vite-plugin-svelte/issues/121

### CODE REVIEW

File: packages/kit/src/core/build/index.js [hunk 1]
```diff
@@ -174,7 +174,10 @@ async function build_client({
 		plugins: [
 			svelte({
 				extensions: config.extensions,
-				emitCss: !config.kit.amp
+				emitCss: !config.kit.amp,
+				compilerOptions: {
+					hydratable: config.kit.ssr
```
Comment (benmccann) [inline packages/kit/src/core/build/index.js:179]: should we use `config.kit.hydrate` instead?
https://kit.svelte.dev/docs#configuration-hydrate

**Review by benmccann — COMMENTED** [review]
`lint` is failing

File: packages/kit/src/core/build/index.js (hunk 1 above)
Comment (dominikg) [reply packages/kit/src/core/build/index.js:179]: good point, somehow i missed that option. should it be `!!(config.kit.ssr && config.kit.hydrate)` ?

without ssr there is nothing to hydrate, right?

Comment (benmccann) [reply packages/kit/src/core/build/index.js:179]: I believe that even without SSR you can still do hydration actually. Basically what it means is we send down the JSON blob with the initial response and then you build the page from that. It saves you from having to do a `fetch` to get the data

Comment (dominikg) [reply packages/kit/src/core/build/index.js]: switched to `!!config.kit.hydrate`

**Comment by dominikg** [comment]
> `lint` is failing

hmm, locally both `pnpm lint` and `pnpm check` passed without failing. 
svelte-check output mentioned a config load issue though for `pnpm check`
```shell
│ Error while loading config
│ TypeError: Cannot read property 'compilerOptions' of undefined
│     at ConfigLoader.loadConfig (/home/dominikg/develop/sveltejs/kit/node_modules/.pnpm/svelte-check@2.2.0_svelte@3.40.0/node_modules/svelte-check/dist/src/index.js:26654:31)
│     at async ConfigLoader.loadAndCacheConfig (/home/dominikg/develop/sveltejs/kit/node_modules/.pnpm/svelte-check@2.2.0_svelte@3.40.0/node_modules/svelte-check/dist/src/index.js:26643:46)
│     at async /home/dominikg/develop/sveltejs/kit/node_modules/.pnpm/svelte-check@2.2.0_svelte@3.40.0/node_modules/svelte-check/dist/src/index.js:26605:17
│     at async Promise.all (index 2)
│     at async ConfigLoader.loadConfigs (/home/dominikg/develop/sveltejs/kit/node_modules/.pnpm/svelte-check@2.2.0_svelte@3.40.0/node_modules/svelte-check/dist/src/index.js:26607:13)
│     at async createLanguageService (/home/dominikg/develop/sveltejs/kit/node_modules/.pnpm/svelte-check@2.2.0_svelte@3.40.0/node_modules/svelte-check/dist/src/index.js:79394:5)
```

**Comment by benmccann** [comment]
lint is passing now. the warning is a bug in `svelte-check`

**Pull request merged**